app/models/infra_conversion_job.rb

```diff
@@ -42,19 +47,77 @@ def load_transitions
     }
   end
 
+  # Example state:
+  #   :state_name => {
+  #     :description => 'State description',
+  #     :weight      => 30,
+  #     :max_retries => 960
+  #   }
   def state_settings
     @state_settings ||= {
+      :collapsing_snapshots => {
+        :description => 'Collapse snapshosts',
+        :weight      => 5,
+        :max_retries => 4.hours / state_retry_interval
+      },
       :running_in_automate => {
-        :max_retries => 8640 # 36 hours with a retry interval of 15 seconds
+        :max_retries => 36.hours / state_retry_interval
       }
     }
   end
 
+  def state_retry_interval
+    @state_retry_interval ||= Settings.transformation.job.retry_interval || 15.seconds
+  end
+
```

Comment:
Just curious, can different states have different retry levels? Is that something that the settings currently support?

Reply:
They could have different retry intervals. This could be added to the `state_settings` in https://github.com/ManageIQ/manageiq/pull/19154. We set it to 15 seconds because that the refresh interval of the UI and we wanted to be in sync.